### ISSUE TITLE
Emissions gcam 5 old new test fails stemming from L111.nonghg_en_R_S_T_Y

### ISSUE THREAD

**Issue by kdorheim** [JGCRI/gcamdata#911]
The `replace_emissions_compdata` branch uses GCAM 5 data output as comparison data. When I try to run the package tests there are so many outputs that fail the old new test that the package tests terminate early (skips L2 old new tests). 

I think that errors in `L111.nonghg_tgej_R_en_S_F_Yh.csv` from `module_emissions_L111.nonghg_en_R_S_T_Y` are being propagated down stream and causing these errors.

I have included a snippet of the old new test errors that are related to the `module_emissions_L111.nonghg_en_R_S_T_Y` output. 

@bpbond Do you think these problems stem from `module_emissions_L111.nonghg_en_R_S_T_Y`  or do you have a different idea about where they could be coming from? Also might be worth while to take a look at `module_emissions_L151.ctrl_R_en_S_T` if there is an obvious place where the row of NMVOC data is being dropped? 

```
test_oldnew.R:143: failure: matches old data system output
sum(olddata[numeric_columns_old]) not equivalent to sum(newdata[numeric_columns_new]).
1/1 mismatches
[1] 813240 - 813246 == -5.93
L111.nonghg_tgej_R_en_S_F_Yh.csv doesn't match (sum test)

test_oldnew.R:133: failure: matches old data system output
dim(olddata) not identical to dim(newdata).
1/2 mismatches
[1] 10081 - 10080 == 1
Dimensions are not the same for L151.nonghg_ctrl_R_en_S_T.csv

test_oldnew.R:143: failure: matches old data system output
sum(olddata[numeric_columns_old]) not equivalent to sum(newdata[numeric_columns_new]).
1/1 mismatches
[1] 729930 - 729809 == 121
L151.nonghg_ctrl_R_en_S_T.csv doesn't match (sum test)

test_oldnew.R:133: failure: matches old data system output
dim(olddata) not identical to dim(newdata).
1/2 mismatches
[1] 6295 - 6291 == 4
Dimensions are not the same for L161.SSP15_EF.csv

test_oldnew.R:146: failure: matches old data system output
round_df(olddata) not equivalent to round_df(newdata).
Different number of rows
L161.SSP15_EF.csv doesn't match

test_oldnew.R:133: failure: matches old data system output
dim(olddata) not identical to dim(newdata).
1/2 mismatches
[1] 6115 - 6111 == 4
Dimensions are not the same for L161.SSP2_EF.csv

test_oldnew.R:146: failure: matches old data system output
round_df(olddata) not equivalent to round_df(newdata).
Different number of rows
L161.SSP2_EF.csv doesn't match

test_oldnew.R:133: failure: matches old data system output
dim(olddata) not identical to dim(newdata).
1/2 mismatches
[1] 6295 - 6291 == 4
Dimensions are not the same for L161.SSP34_EF.csv
```

**Comment by bpbond** [comment]
When I run the driver up to (but not including) `module_emissions_L111.nonghg_en_R_S_T_Y`, I get an oldnew failure for `L111.RsrcCurves_EJ_R_Ffos`. Odd.

**Comment by bpbond** [comment]
My guess (can't dig further right now, but should have time later today) is that the problem centers on lines 236-244 in `L111.nonghg_en_R_S_T_Y`:
```R
    L111.nonghg_tg_R_en_S_F_Yh %>%
      left_join(L101.in_EJ_R_en_S_F_Yh.mlt, by = c("GCAM_region_ID", "supplysector", "subsector", "stub.technology", "year")) %>%
      # the old data system was doing a straight division of some _tiny_ numbers here, which made it impossible to replicate
      # we now round; see
      mutate(value = round(value, 8) / round(energy, 8)) %>%
      ungroup %>%
      select(-energy) %>%
      replace_na(list(value = 0)) ->
      L111.nonghg_tgej_R_en_S_F_Yh   # OUTPUT
```
As the comment says, this tiny-number division caused us problems before, and it may be doing so again.

**Comment by kdorheim** [comment]
It is really odd that `L111.RsrcCurves_EJ_R_Ffos.csv` fails the old new test. It fails when I replace the energy comp data but that should not be happening on this branch... 

 Do you think that the `L111.nonghg_tgej_R_en_S_F_Yh` numeric discrepancies are different enough to cause concern or can we change the test flags so that it can pass along?

**Comment by bpbond** [comment]
Ugh, I don't know. It _looks_ like the differences are all absolutely tiny but some are relatively large. Let me make another graph looking at this.

**Comment by bpbond** [comment]
Most of the data in `L111.nonghg_tgej_R_en_S_F_Yh` have zero or tiny oldnew relative differences:
![reldiff1](https://user-images.githubusercontent.com/1956468/38316793-88b5abbe-37f9-11e8-871e-8b067fbb8985.png)

But small values have larger relative errors, and in particular there's a whole group that's become zero:
![reldiff2](https://user-images.githubusercontent.com/1956468/38316697-4a73fc2a-37f9-11e8-948d-795740fb896f.png)

**Comment by bpbond** [comment]
@kdorheim So, I don't know. Should we just try seeing if these changes screw up things downstream?

**Comment by bpbond** [comment]
Addressed by #935 .
